.github/workflows/build-deps.yml

```diff
@@ -0,0 +1,196 @@
+        default: false
+        type: boolean
+      llvm_build_cross_compilation:
+        description: 'LLVM Webassembly arm64 (cross compilation)'
+        required: false
+        default: false
+        type: boolean
+      emscripten_build:
+        description: 'Emscripten'
+        required: false
+        default: false
+        type: boolean
+      nodejs_build:
+        description: 'Node.js arm64'
+        required: false
+        default: false
+        type: boolean
+      emsdk_build:
+        description: 'emsdk arm64'
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  build-llvm-webassembly-arm64:
+    if: ${{ inputs.llvm_build }} ||  ${{ inputs.emsdk_build }}
+
+    runs-on: [self-hosted, Windows, ARM64, WASM]
+    timeout-minutes: 600
+
+    steps:
+    
+    - name: Build LLVM WebAssembly arm64
+      shell: cmd
+      run: |
+        exit
+        git clone --config core.autocrlf=false https://github.com/llvm/llvm-project.git --single-branch --depth 1
+        cd llvm-project
+        if exist build_arm64 rmdir /s /q build_arm64
+        call "%ProgramFiles%\Microsoft Visual Studio\2022\Preview\VC\Auxiliary\Build\vcvarsall.bat" arm64
+        cmake -G Ninja -S llvm -B build_arm64 ^
+          -DCLANG_ENABLE_ARCMT=OFF ^
+          -DCLANG_ENABLE_STATIC_ANALYZER=OFF ^
+          -DCMAKE_BUILD_TYPE=Release ^
+          -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON ^
+          -DCMAKE_CXX_COMPILER_LAUNCHER='ccache' ^
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ^
+          -DCMAKE_LINKER="C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\Llvm\x64\bin\lld-link.exe" ^
+          -DLLVM_BUILD_LLVM_DYLIB=OFF ^
+          -DLLVM_LINK_LLVM_DYLIB=OFF ^
+          -DLLVM_DISTRIBUTION_COMPONENTS='clang;lld;llvm-ar;llvm-addr2line;llvm-cxxfilt;llvm-dwarfdump;llvm-dwp;llvm-nm;llvm-objcopy;llvm-objdump;llvm-ranlib;llvm-readobj;llvm-size;llvm-strings;llvm-strip;llvm-symbolizer;clang-resource-headers' ^
+          -DLLVM_ENABLE_ASSERTIONS=OFF ^
+          -DLLVM_ENABLE_BINDINGS=OFF ^
+          -DLLVM_ENABLE_LIBXML2=OFF ^
+          -DLLVM_ENABLE_PROJECTS='lld;clang' ^
+          -DLLVM_ENABLE_TERMINFO=ON ^
+          -DLLVM_INCLUDE_EXAMPLES=OFF ^
+          -DLLVM_INCLUDE_TESTS=OFF ^
+          -DLLVM_INSTALL_TOOLCHAIN_ONLY=ON ^
+          -DLLVM_TARGETS_TO_BUILD='host;WebAssembly' ^
+          -DLLVM_TOOL_LTO_BUILD=OFF ^
+          -DLLVM_TOOLCHAIN_TOOLS='clang;lld;llvm-ar;llvm-addr2line;llvm-cxxfilt;llvm-dwarfdump;llvm-dwp;llvm-nm;llvm-objcopy;llvm-objdump;llvm-ranlib;llvm-readobj;llvm-size;llvm-strings;llvm-strip;llvm-symbolizer;clang-resource-headers' ^
+          -DLLVM_USE_CRT_RELEASE=MT ^
+          -DLLVM_USE_CRT_DEBUG=MTd 
+        cd build_arm64
+        ninja -v install-distribution
+
+    - name: Pack llvm-arm64.zip
+      run: |
+        cd llvm-project/build_arm64
+        $unneeded_tool = 'clang-check', 'clang-cl', 'clang-cpp', 
+          'clang-extdef-mapping', 'clang-format',
```

Comment:
Since those are not needed, are you sure that they needs to be compiled in the first place?

Reply:
replied above

Reply:
I am not sure where this question was answered? I am asking whether you are sure that there is no other option do disable building of these when we don't need them eventually or some further work in that regard might be done?

Reply:
It looks like there is no way to exclude them from building. emscripten release pipeline does it the same way.